MonstieWash/Assets/Scripts/ScriptedEvents/ScriptedEventManager.cs

```diff
@@ -0,0 +1,103 @@
+
+    private void Awake()
+    {
+        //Ensure there's only one
+        if (Instance == null) Instance = this;
+        else Destroy(this);
+
+        m_monsterBrain = FindFirstObjectByType<MonsterBrain>();
+    }
+
+    public void Subscribe(GameObject subscriber, params ScriptedEventType[] permissions)
+    {
+        if (!m_subscribers.ContainsKey(subscriber))     // subscriber can only be subscribed once
+        {
+            // create and add new subscriber
+            Dictionary<ScriptedEventType, bool> myPerms = new Dictionary<ScriptedEventType, bool>();
+            m_subscribers.Add(subscriber, myPerms);
+
+            // create and add perms for new subscriber
+            foreach (ScriptedEventType eventType in Enum.GetValues(typeof(ScriptedEventType)))
+            {
+                bool found = false;
+                foreach (var permission in permissions)
+                {
+                    if (permission == eventType)
+                    {
+                        m_subscribers[subscriber].Add(eventType, true);
+                        found = true;
+                    }
+                }
+                if (!found) m_subscribers[subscriber].Add(eventType, false);
+            }
+        }
+    }
+
+    public void Unsubscribe(GameObject subscriber)
+    {
+        if (m_subscribers.ContainsKey(subscriber)) m_subscribers.Remove(subscriber);
+    }
+
+    public void ModifyPermission(GameObject subscriber, ScriptedEventType type, bool newVal)
+    {
+        if (m_subscribers.ContainsKey(subscriber))
+        {
+            m_subscribers[subscriber][type] = newVal;
+        }
+    }
+
+    public void RunScriptedEvent(GameObject caller, ScriptedEventType type)
+    {
+        switch (type)
+        {
+            case ScriptedEventType.SetAngry:
+                SetAngry(caller);
+                break;
+            default:
+                break;
+        }
+    }
+
+    public void SetAngry(GameObject caller)
+    {
+        if (m_subscribers.ContainsKey(caller))  // if caller is subscribed
+        {
+            if (m_subscribers[caller][ScriptedEventType.SetAngry])  // if caller has permission to execute this scripted event
+            {
+                // currently sets the monster to its maximum anger
+                float maxAnger = monsterAngryMood.MoodUpperLimit;
+                m_monsterBrain.UpdateMood(maxAnger, monsterAngryMood);
+            }
+            else
+            {
+                if (debugMode) Debug.LogWarning($"{caller.name} tried to call scripted event SetAngry but is missing permissions to do so!");
+            }
+        }
+        else
+        {
+            if (debugMode) Debug.LogWarning($"{caller.name} tried to call scripted event SetAngry but hasn't subscribed to Scripted Events!");
+        }
+    }
```

Comment:
💡 **Suggestion** | _This is a suggestion, feel free to take it or leave it._

I think it would help with the cohesion of this method to have all the subscriber and permission checking be done in the `RunScriptedEvent` method (within the case section I'd imagine). That way `SetAngry` only has to worry about actually setting the mood to angry and none of the overhead.